sdk/appplatform/mgmt/src/main/java/com/azure/resourcemanager/appplatform/implementation/SpringAppDeploymentImpl.java

```diff
@@ -0,0 +1,446 @@
+        }
+        return inner().properties().active();
+    }
+
+    @Override
+    public OffsetDateTime createdTime() {
+        if (inner().properties() == null) {
+            return null;
+        }
+        return inner().properties().createdTime();
+    }
+
+    @Override
+    public List<DeploymentInstance> instances() {
+        if (inner().properties() == null) {
+            return null;
+        }
+        return inner().properties().instances();
+    }
+
+    @Override
+    public String getLogFileUrl() {
+        return getLogFileUrlAsync().block();
+    }
+
+    @Override
+    public Mono<String> getLogFileUrlAsync() {
+        return manager().inner().getDeployments().getLogFileUrlAsync(
+            parent().parent().resourceGroupName(), parent().parent().name(), parent().name(), name()
+        )
+            .map(LogFileUrlResponseInner::url);
+    }
+
+    private void ensureDeploySettings() {
+        if (inner().properties() == null) {
+            inner().withProperties(new DeploymentResourceProperties());
+        }
+        if (inner().properties().deploymentSettings() == null) {
+            inner().properties().withDeploymentSettings(new DeploymentSettings());
+        }
+    }
+
+    private void ensureSource() {
+        if (inner().properties() == null) {
+            inner().withProperties(new DeploymentResourceProperties());
+        }
+        if (inner().properties().source() == null) {
+            inner().properties().withSource(new UserSourceInfo());
+        }
+    }
+
+    private File compressSource(File sourceFolder) throws IOException {
+        File compressFile = File.createTempFile("java_package", "tar.gz");
+        try (TarArchiveOutputStream tarArchiveOutputStream = new TarArchiveOutputStream(
+                 new GZIPOutputStream(new FileOutputStream(compressFile)))) {
+            tarArchiveOutputStream.setLongFileMode(TarArchiveOutputStream.LONGFILE_GNU);
+
+            for (Path sourceFile : Files.walk(sourceFolder.toPath()).collect(Collectors.toList())) {
+                String relativePath = sourceFolder.toPath().relativize(sourceFile).toString();
+                TarArchiveEntry entry = new TarArchiveEntry(sourceFile.toFile(), relativePath);
+                if (sourceFile.toFile().isFile()) {
+                    try (InputStream inputStream = new FileInputStream(sourceFile.toFile())) {
+                        tarArchiveOutputStream.putArchiveEntry(entry);
+                        IOUtils.copy(inputStream, tarArchiveOutputStream);
+                        tarArchiveOutputStream.closeArchiveEntry();
+                    }
+                } else {
+                    tarArchiveOutputStream.putArchiveEntry(entry);
+                    tarArchiveOutputStream.closeArchiveEntry();
+                }
+            }
+        }
+        return compressFile;
+    }
+
+    private Mono<ShareFileAsyncClient> createShareFileAsyncClient(ResourceUploadDefinition option, long maxSize) {
```

Comment:
Why `Mono`? Normally you should create `ShareFileAsyncClient`, which is not async, and pass it to other methods. Async call of `create(maxSize)` appear not about the `createShareFileAsyncClient` method.

Reply:
Since all callers need the `create` function, so I add it to the common function. Unless we do `create` repeatedly in every caller, we'll always need a function to return a `Mono<ShareFileAsyncClient>` as common function. I could remove this due to it is little repeated code.

Reply:
You need 2 things from this mtehod, the `ShareFileAsyncClient` which is just an object, and a `Mono<Void>` as completion of the `create`. You should not mix the 2 to a `Mono<ShareFileAsyncClient>`, it just abuses the reactor.